### DISCUSSION TITLE
ENV Variables not working

### DISCUSSION THREAD

**Discussion by jd-apprentice** [vercel/next.js#37099] [Help] [answered]
Hello!, I'm asking for help since I've trying for a long time and I cant make it work, I've downloaded this example -> [mongodb + mongoose](https://github.com/vercel/next.js/tree/canary/examples/with-mongodb-mongoose)

right now I have the problem that .env.local is not being counted? this line of code only works if I literally delete the .env.local and hardcode the uri

```const MONGODB_URI = process.env.MONGODB_URI```

in case is not clear I'm completely new to next I'm trying to get to work that example since it got my interest but Idk if docs are outdated or i'm missing something

also just in case I'dont want to use NEXT_PUBLIC since I want to keep secret the mongodb_uri since is my actual db which I use for some projects

edit: I made a friend tried the same example and it worked for him, what is messed up with me idk

**Comment by jd-apprentice** [comment, marked as answer]
I've found the problem my MONGODB_URI had a special caracter and I had to use \ to escape it, anyways i'm glad it works now lmao

**Reply by JoMind** [reply]
Hey, can't make it work no matter what, could you add extra detail what helped you to make ENV variables work for this mongodb + mongoose example? Thanks!

**Reply by jd-apprentice** [reply]
Hey! @JoMind my password had a special character that was `$` I added an escape character in this case `\` before the pass for example ->

`myPa\$\$word`
